lib/internal/Magento/Framework/View/Layout.php

```diff
@@ -1138,18 +1143,22 @@ protected function _prepareMessageGroup($messageGroups)
      */
     public function isCacheable()
     {
-        $this->build();
-        $elements = $this->getXml()->xpath('//' . Element::TYPE_BLOCK . '[@cacheable="false"]');
-        $cacheable = $this->cacheable;
-        foreach ($elements as $element) {
-            $blockName = $element->getBlockName();
-            if ($blockName !== false && $this->structure->hasElement($blockName)) {
-                $cacheable = false;
-                break;
+        if (!isset($this->isCacheableCache)) {
```

Comment:
Why not using use comparator here? `$this->isCacheableCache === null`
We are sure that isCacheableCache property exists, so we can avoid this check